src/components/Sections/GoogleMapsSection/index.tsx

```diff
@@ -0,0 +1,179 @@
+import { Loader } from "@googlemaps/js-api-loader";
+// eslint-disable-next-line
+const markerIcon = require("../../../assets/Marker.png");
+
+const DEFAULT_STYLE: google.maps.MapTypeStyle[] = [
+  {
+    featureType: "water",
+    elementType: "geometry",
+    stylers: [{ color: "#e9e9e9" }, { lightness: 17 }],
+  },
+  {
+    featureType: "landscape",
+    elementType: "geometry",
+    stylers: [{ color: "#f5f5f5" }, { lightness: 20 }],
+  },
+  {
+    featureType: "road.highway",
+    elementType: "geometry.fill",
+    stylers: [{ color: "#ffffff" }, { lightness: 17 }],
+  },
+  {
+    featureType: "road.highway",
+    elementType: "geometry.stroke",
+    stylers: [{ color: "#ffffff" }, { lightness: 29 }, { weight: 0.2 }],
+  },
+  {
+    featureType: "road.arterial",
+    elementType: "geometry",
+    stylers: [{ color: "#ffffff" }, { lightness: 18 }],
+  },
+  {
+    featureType: "road.local",
+    elementType: "geometry",
+    stylers: [{ color: "#ffffff" }, { lightness: 16 }],
+  },
+  {
+    featureType: "poi",
+    elementType: "geometry",
+    stylers: [{ color: "#f5f5f5" }, { lightness: 21 }],
+  },
+  {
+    featureType: "poi.park",
+    elementType: "geometry",
+    stylers: [{ color: "#dedede" }, { lightness: 21 }],
+  },
+  {
+    elementType: "labels.text.stroke",
+    stylers: [{ visibility: "on" }, { color: "#ffffff" }, { lightness: 16 }],
+  },
+  {
+    elementType: "labels.text.fill",
+    stylers: [{ saturation: 36 }, { color: "#333333" }, { lightness: 40 }],
+  },
+  { elementType: "labels.icon", stylers: [{ visibility: "off" }] },
+  {
+    featureType: "transit",
+    elementType: "geometry",
+    stylers: [{ color: "#f2f2f2" }, { lightness: 19 }],
+  },
+  {
+    featureType: "administrative",
+    elementType: "geometry.fill",
+    stylers: [{ color: "#fefefe" }, { lightness: 20 }],
+  },
+  {
+    featureType: "administrative",
+    elementType: "geometry.stroke",
+    stylers: [{ color: "#fefefe" }, { lightness: 17 }, { weight: 1.2 }],
+  },
+] as google.maps.MapTypeStyle[];
+
+@Component({
+  name: "GoogleMapsSection",
+})
+class GoogleMapsSection extends BaseComponent<GoogleMapsSectionProps> {
+  map: google.maps.Map | null = null;
+
+  @Prop() title: GoogleMapsSectionProps["title"];
+  @Prop({ required: false }) apikey!: GoogleMapsSectionProps["apikey"];
+  @Prop({ required: false })
+  startLocation!: GoogleMapsSectionProps["startLocation"];
+  @Prop({ required: false, default: 15 }) zoom!: GoogleMapsSectionProps["zoom"];
+  @Prop({ required: false, default: "en" })
+  language!: GoogleMapsSectionProps["language"];
+  @Prop({ required: false }) locations!: GoogleMapsSectionProps["locations"];
+  @Prop({ required: false, default: "Contact Us" })
+  buttonLabel!: GoogleMapsSectionProps["buttonLabel"];
+  @Prop({ required: true })
+  handleButtonClick!: GoogleMapsSectionProps["handleButtonClick"];
+
+  placeMarkers(map: google.maps.Map, ...locations: MapsLocation[]): void {
```

Comment:
We should change the signature to this, so we don't need to destruct the array:
```suggestion
  placeMarkers(map: google.maps.Map, locations: MapsLocation[]): void {
```
Please have a look at my other comment on this file: